### ISSUE TITLE
[Feature request] NixOS tests screenshot verification system

### ISSUE THREAD

**Issue by ericsagnes** [NixOS/nixpkgs#26009]
In NixOS tests, Screenshots can be taken with the `screenshot` command, but there is no way to check the correctness of a screenshot.

[OpenQA](http://open.qa/), the test system from open suse, has a very interesting [needle](http://open.qa/docs/#_needles) mechanism that allow to check if a screenshot is correct by comparing sections of it with a reference.
Having a similar mechanism available in NixOS tests would greatly improve how graphic software and features are tested.

**Comment by copumpkin** [comment]
cc @aszlig who added the current OCR screenshot mechanism we have in place today.

**Comment by aszlig** [comment]
Yep, not only the needle system but also their [UI](https://openqa.opensuse.org/tests/406918#step/inkscape/8) is really great. Having a similar needle system would also improve our OCR detection.
